src/Microsoft.Extensions.Logging.Filter/FilterLoggerSettings.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections;
+using System.Collections.Generic;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.Extensions.Logging
+{
+    /// <summary>
+    /// Filter settings for messages logged by an <see cref="ILogger"/>.
+    /// </summary>
+    public class FilterLoggerSettings : IFilterLoggerSettings, IEnumerable<KeyValuePair<string, LogLevel>>
+    {
+        IChangeToken IFilterLoggerSettings.ChangeToken => null;
+
+        public IDictionary<string, LogLevel> Switches { get; set; } = new Dictionary<string, LogLevel>();
```

Comment:
With VB.NET using case-insensitive class/namespace names, an insensitive comparer might be called for here, just to improve usability. Maybe a nit, hard to change later however.